src/mds/MDCache.cc

```diff
@@ -6542,13 +6548,17 @@ void MDCache::trim_lru(uint64_t count, expiremap& expiremap)
           << " pinned=" << lru.lru_get_num_pinned()
           << dendl;
 
-  for (;;) {
+  bool throttled = false;
+  while (1) {
+    throttled |= (size_t)trim_counter.get() >= trim_threshold;
```

Comment:
DecayCounter::get() is expensive. we should calculate nr_to_trim outside the loop.

Reply:
done